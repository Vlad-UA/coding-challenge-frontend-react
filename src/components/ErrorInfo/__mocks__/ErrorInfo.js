// Core
import React from 'react';

const ErrorInfoMock = () => <div className="error-info-mock">ErrorInfoMock</div>;
ErrorInfoMock.displayName = 'ErrorInfoMock';
export default ErrorInfoMock;
