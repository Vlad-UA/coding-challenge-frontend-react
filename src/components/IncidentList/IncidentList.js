// Core
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

// Components
import IncidentCard from '../IncidentCard/IncidentCard';
import Pagination from '../Pagination/Pagination';

// Instruments
import styles from './IncidentList.module.scss';

export default class IncidentList extends PureComponent {
  render() {
    const {
      openPage, incidentsData, currentPage, incidentsData: { incidents = [] },
    } = this.props;

    console.log('incidentsData', incidentsData);

    return (
      <div className={styles.incidentList}>
        {
          incidents.length > 0
            ? (
              <>
                {incidents.map(
                  incident => (
                    <div className={styles.incidentCard} key={incident.id}>
                      <IncidentCard incident={incident} />
                    </div>
                  ),
                )}
                <Pagination currentPage={currentPage} openPage={openPage} />
              </>
            )
            : <p>No results</p>
        }
      </div>
    );
  }
}

IncidentList.propTypes = {
  incidentsData: PropTypes.shape({
    incidents: PropTypes.arrayOf(
      PropTypes.object,
    ),
  }).isRequired,
  currentPage: PropTypes.number.isRequired,
};
