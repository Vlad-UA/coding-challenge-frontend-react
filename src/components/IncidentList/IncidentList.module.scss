.incidentList {

  .total {
    display: grid;
    justify-items: end;
    margin-top: 40px;
  }

  .incidentCard {
    margin: 10px;
  }
}
