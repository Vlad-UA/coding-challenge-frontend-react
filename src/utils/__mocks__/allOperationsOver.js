export const allOperationsOver = () => new Promise(resolve => setTimeout(resolve, 0));
