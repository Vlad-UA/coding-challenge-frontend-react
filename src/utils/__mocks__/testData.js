const OCCURRED_AT = 1552325609;

class IncidentData {
  constructor(
    id = Math.floor(10000 + Math.random() * 90000),
    title = `Title ${id}`,
    description = `Description ${id}`,
    address = `Address ${id}`,
    occurred_at = OCCURRED_AT,
    updated_at = 1553542734,
    image_url_thumb = 'https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg',
    type = 'Theft',
  ) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.address = address;
    this.occurred_at = occurred_at;
    this.updated_at = updated_at;
    this.media = {
      image_url_thumb,
    };
    this.type = type;
  }
}

// One Incident
export class IncidentModel {
  constructor({
    id,
    title,
    description,
    address,
    occurred_at,
    updated_at,
    image_url_thumb,
    type,
  }) {
    this.incident = new IncidentData(
      id,
      title,
      description,
      address,
      occurred_at,
      updated_at,
      image_url_thumb,
      type,
    );
  }
}

// Many Incidents
export class IncidentsModel {
  constructor(
    ids = [97572, 94547],
  ) {
    this.incidents = ids.map(id => new IncidentData(id));
  }
}


class LocationData {
  constructor(
    id = Math.floor(10000 + Math.random() * 90000),
    occurred_at = OCCURRED_AT,
  ) {
    this.properties = {
      id,
      occurred_at,
    };
    this.geometry = {
      type: 'Point',
      coordinates: [
        -116.1960257,
        43.6334557,
      ],
    };
  }
}

export class LocationsModel {
  constructor(
    ids = [97572],
  ) {
    this.type = 'FeatureCollection';
    this.features = ids.map(id => new LocationData(id));
  }
}
