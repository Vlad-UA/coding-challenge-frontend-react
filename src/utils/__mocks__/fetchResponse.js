// Test utilities
import { IncidentsModel } from './testData';

export class FetchResponse {
  constructor({
    status = global.fetchResponseStatus,
    fetchResponseData = new IncidentsModel(),
  } = {}) {
    return {
      status,
      json() {
        return Promise.resolve(fetchResponseData);
      },
    };
  }
}
