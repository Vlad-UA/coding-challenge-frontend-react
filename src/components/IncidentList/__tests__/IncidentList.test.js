/* eslint-env jest */

// Core
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

// Test utilities
import renderer from 'react-test-renderer';
import { IncidentsModel } from '../../../utils/__mocks__/testData';

// Component
import IncidentList from '../IncidentList';

jest.mock('../../IncidentCard/IncidentCard', () => () => 'IncidentCard');
jest.mock('../../Pagination/Pagination', () => () => 'Pagination');

describe('Component IncidentList:', () => {
  test('Must match snapshot -- with some incidents:', () => {
    const { incidents } = new IncidentsModel();

    const incidentList = renderer
      .create(
        <BrowserRouter>
          <IncidentList
            incidents={incidents}
            currentPage={1}
            openPage={jest.fn}
          />
        </BrowserRouter>,
      )
      .toJSON();

    expect(incidentList).toMatchSnapshot();
  });

  test('Must match snapshot -- without incidents:', () => {
    const incidentList = renderer
      .create(
        <BrowserRouter>
          <IncidentList
            incidents={[]}
            currentPage={1}
            openPage={jest.fn}
          />
        </BrowserRouter>,
      )
      .toJSON();

    expect(incidentList).toMatchSnapshot();
  });
});
