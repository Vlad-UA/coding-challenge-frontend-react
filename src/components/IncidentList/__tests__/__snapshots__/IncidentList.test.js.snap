// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component IncidentList: Must match snapshot -- with some incidents: 1`] = `
<div
  className="incidentList"
>
  <div
    className="total"
  >
    total: 2
  </div>
  <div
    className="incidentCard"
  >
    IncidentCard
  </div>
  <div
    className="incidentCard"
  >
    IncidentCard
  </div>
  Pagination
</div>
`;

exports[`Component IncidentList: Must match snapshot -- without incidents: 1`] = `
<div
  className="incidentList"
>
  <p>
    No results
  </p>
</div>
`;
